eagle-stores/src/context/auth/auth-context.js

```diff
@@ -1,39 +1,51 @@
 import { createContext, useContext, useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { loginServices } from '../../services/Services';
+import { loginServices, SignupServices } from '../../services/Services';
+import { useData } from '../data/data-context';
 
 const AuthContext = createContext();
 
 const AuthProvider = ({ children }) => {
   const navigate = useNavigate();
   const localStorageToken = JSON.parse(localStorage.getItem('login'));
   const [token, setToken] = useState(localStorageToken?.token);
-  const [user, setUser] = useState(null);
+  const localStorageUser = JSON.parse(localStorage.getItem('user'));
+  const [user, setUser] = useState(localStorageUser?.user);
 
   useEffect(() => {
     const fetchToken = JSON.parse(localStorage.getItem('login'));
     if (fetchToken) setToken(fetchToken.tokens);
   }, []);
 
-  const loginHandler = async (email, password) => {
+  const signupUser = async (email, password, name) => {
     try {
-      const resp = await loginServices({ email, password });
-      if (resp.status === 200 || resp.status === 201) {
+      const resp = await SignupServices({ email, password, name });
+      if (resp.status === 201) {
         localStorage.setItem(
           'login',
-          JSON.stringify({ tokens: resp.data.encodedToken })
+          JSON.stringify({
+            token: resp.data.encodedToken,
+            user: resp.data.createdUser,
+          })
         );
-        setUser(resp.data.foundUser);
+        setUser(resp.data.createdUser);
         setToken(resp.data.encodedToken);
-        navigate('/product');
       }
-    } catch (err) {
-      console.log(err);
+    } catch (error) {
+      console.log(error);
```

Comment:
instead of consoling you can display the error on screen

Reply:
Yeah I am going to use Toast for that